### PR TITLE
fix: [PIE-5621]: added edit functionality for a barrier name in the pipeline's flow control section

### PR DESCRIPTION
**Pull request by Eric761**
### Summary

Added edit functionality for a barrier name in the pipeline's flow control section 

#### Demo

Before:

image-20221001-172824

After:

https://user-images.githubusercontent.com/73639897/193597479-123e765b-ca3e-4e22-9152-bdad3a827ee8.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 6 Issues 
* Code issue 6 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11633)

**Comment by Eric761** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/FlowControl.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ export interface BarrierListProps {
   list: Array<Barrier>
   createItem: (push: (data: Barrier) => void) => void
   deleteItem: (index: number, remove: (a: number) => void) => void
-  commitItem: (data: Barrier, index: number) => void
+  commitItem: (data: Barrier, index: number, barrierMode: 'EDIT' | null) => void
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/components/PipelineStudio/FlowControl/FlowControl.tsx:78]: Make it `barrierMode?: 'EDIT'`

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/FlowControl.tsx [hunk 2]
```diff
@@ -141,14 +141,14 @@ export function FlowControl(
     updateBarriers(updatedState)
   }
 
-  const commitBarrier = (barrierData: Barrier, index: number): void => {
+  const commitBarrier = (barrierData: Barrier, index: number, barrierMode: 'EDIT' | null): void => {
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/components/PipelineStudio/FlowControl/FlowControl.tsx:144]: Make it take default value of null like this `barrierMode: Barrier['mode'] = null`

**Review by kapilkumawat86 — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/FlowControl.tsx [hunk 3]
```diff
@@ -356,9 +373,10 @@ export function BarrierList({
                                     inputGroup: {
                                       onBlur: () =>
                                         !formik.errors?.barriers?.[index] &&
-                                        commitItem(formik.values.barriers[index], index)
+                                        commitItem(formik.values.barriers[index], index, null)
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/components/PipelineStudio/FlowControl/FlowControl.tsx:376]: No need to pass `null` if you use default param value

**Review by depanshu-dhiman — COMMENTED** [review]
Please address

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/FlowControl.tsx (hunk 3 above)
Comment (Eric761) [reply src/modules/70-pipeline/components/PipelineStudio/FlowControl/FlowControl.tsx]: Sure

**Review by depanshu-dhiman — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**